metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/AggregationHelper.java

```diff
@@ -63,173 +48,33 @@ public static Optional<Metrics> getAggregatedMetrics(ExecutionsRequestBody allSu
-
-                    latestValidationExecutionForVersion.ifPresent(validationExecution -> {
-                        ValidatorVersionInfo validatorVersionInfo = new ValidatorVersionInfo()
-                                .name(validatorVersionName)
-                                .isValid(validationExecution.isIsValid())
-                                .dateExecuted(validationExecution.getDateExecuted())
-                                .numberOfRuns(validatorVersionExecutions.size())
-                                .passingRate(getPassingRate(validatorVersionExecutions));
-
-                        if (!validationExecution.isIsValid() && StringUtils.isNotBlank(validationExecution.getErrorMessage())) {
-                            validatorVersionInfo.errorMessage(validationExecution.getErrorMessage());
-                        }
-
-                        validatorVersionNameToVersionInfo.put(validatorVersionName, validatorVersionInfo);
-                    });
-                });
-
-                // Set validation info for the validator tool
-                ValidatorInfo validatorInfo = new ValidatorInfo()
-                        .mostRecentVersionName(validatorVersionNameToVersionInfo.get(latestValidationExecution.get().getValidatorToolVersion()).getName())
-                        .validatorVersions(validatorVersionNameToVersionInfo.values().stream().toList())
-                        .numberOfRuns(validatorToolExecutions.size())
-                        .passingRate(getPassingRate(validatorToolExecutions));
-
-                validatorToolToValidationInfo.put(validatorTool.toString(), validatorInfo);
-            }
-        });
-
-        // This shouldn't happen because all validation executions should the required fields, but check anyway
-        if (validatorToolToValidationInfo.isEmpty()) {
-            return Optional.empty();
-        }
-        return Optional.of(new ValidationStatusMetric().validatorTools(validatorToolToValidationInfo));
-    }
-
-    static Optional<ValidationExecution> getLatestValidationExecution(List<ValidationExecution> executions) {
-        if (executions.isEmpty()) {
-            return Optional.empty();
-        }
-
-        boolean containsInvalidDate = executions.stream().anyMatch(execution -> checkExecutionDateISO8601Format(execution.getDateExecuted()).isEmpty());
-        if (containsInvalidDate) {
-            return Optional.empty();
+    public static Optional<Metrics> getAggregatedMetrics(List<Metrics> aggregatedMetrics) {
+        Metrics overallMetrics = new Metrics();
+        // Set run metrics
+        Optional<ExecutionStatusMetric> aggregatedExecutionStatus = new ExecutionStatusAggregator().getAggregatedMetricFromMetricsList(aggregatedMetrics);
+        boolean containsRunMetrics = aggregatedExecutionStatus.isPresent();
+        if (aggregatedExecutionStatus.isPresent()) {
```

Comment:
The data is such that if we can't aggregate an `ExecutionStatusMetric`, we can't aggregate these other things, either?

Reply:
Yeah, it was previously like this. The idea is that execution status is a metric that we require from all run executions thus ExecutionStatusMetric is an aggregated metric that the webservice requires